src/sasl/cyrsasl_external.erl

```diff
@@ -45,9 +45,10 @@ stop() ->
 
 -spec mech_new(Host :: ejabberd:server(),
                Creds :: mongoose_credentials:t()) -> {ok, sasl_external_state()}.
-mech_new(_Host, Creds) ->
+mech_new(Host, Creds) ->
     Cert = mongoose_credentials:get(Creds, client_cert, no_cert),
-    maybe_extract_certs(Cert, Creds).
+    NewCreds = mongoose_credentials:extend(Creds, [{server, Host}]),
```

Comment:
I'm pretty sure that the information is already in Creds, did you check it? You can read it with `mongoose_credentials:lserver/1`.